### PR TITLE
Added solution for pattern print using recursion

### PR DESCRIPTION
**Pull request by chaltidutta**
Added this code to the project. Kindly check if everything is in proper order.

 ✅ Code is added in right directory(Folder).
 ✅ Brief Description of Solution is given.
 ✅ Time and Space Complexity is mentioned
 ✅ Code is well indented.
 ✅ Comments are clearly written wherever necessary
 ✅ Code is successfully compiled and passes all test cases

### CODE REVIEW

File: 2. String/binary_add/binaryAddition.cpp [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+    } 
+    cout<<result<<endl; //print the ans
+}
+
+int main() {
+    int t;
+    cin>>t;
+    while(t-->0){
+        string a,b;
+        cin>>a>>b;
```
Comment (anjali1903) [inline 2. String/binary_add/binaryAddition.cpp]: Take these inputs in the solve function itself.

**Comment by anjali1903** [comment]
Also, please change the name of the file to binary_addition_yourname.cpp

File: 2. String/binary_add/binary_addition_chalti.cpp [hunk 2]
```diff
@@ -11,32 +11,31 @@
...
-        x.size()--; y.size()--;  //increase the digit number or move to the next digit
-    } 
-    cout<<result<<endl; //print the ans
-}
-
-int main() {
+void solve() { 
     int t;
     cin>>t;
     while(t-->0){
```
Comment (anjali1903) [inline 2. String/binary_add/binary_addition_chalti.cpp]: take the test case in the main function and rest of the code in solve function. please refer to the template in Contributing.md

File: 2. String/binary_add/binary_addition_chalti.cpp [hunk 3]
```diff
@@ -11,32 +11,31 @@
...
+        int sumOfdigit = 0;  //initialize the sum of the digit as zero 
+
+        //traverse both of the strings from last digit to first digit
+        while (x.size() >= 0 || y.size() >= 0 || sumOfdigit == 1) 
+        { 
+            sumOfdigit += ((x.size() >= 0)? x[x.size()] - '0': 0); 
+            sumOfdigit += ((y.size() >= 0)? y[y.size()] - '0': 0); 
+            result = char(sumOfdigit % 2 + '0') + result; //if current result if 1 or 0 then add it with the resultant string
+            sumOfdigit /= 2;  //carry
+            x.size()--; y.size()--;  //increase the digit number or move to the next digit
```
Comment (anjali1903) [inline 2. String/binary_add/binary_addition_chalti.cpp]: I'm not sure if you can decrement the size of a string directly from here. You can take the size in a variable and decrement that.

Comment (chaltidutta) [reply 2. String/binary_add/binary_addition_chalti.cpp]: I have fixed it, now you can check .

**Comment by anjali1903** [comment]
Good work!

**Pull request merged**